tests/rig/coverage.py

```
@@ -0,0 +1,36 @@
+        "COVERAGE_FILE": str(cov_file),
+    }
+
+
+def combine_and_report(reports_dir: Path) -> None:
+    """Combine all per-group ``.coverage.<group>`` files and emit xml + html."""
+    if not any(reports_dir.glob(".coverage.*")):
+        return
+    target = reports_dir / ".coverage"
+    if target.exists():
+        target.unlink()
+    subprocess.run(
```

Ruff (0.15.13)

[error] `subprocess` call: check for execution of untrusted input

(S603)